rust/src/cli/service.rs

```diff
@@ -99,3 +111,86 @@ fn relocate_file(file_path: &Path) -> Result<(), CliError> {
     );
     Ok(())
 }
+
+fn pin_iface_name(cfg_dir: &Path) -> Result<(), CliError> {
+    let file_path = cfg_dir.join(PIN_STATE_FILENAME);
+    let mut fd = std::fs::File::open(&file_path)?;
+    let mut content = String::new();
+    fd.read_to_string(&mut content)?;
+    let pin_state: NetworkState = serde_yaml::from_str(&content)?;
+    let mut cur_state = NetworkState::new();
+    cur_state.set_kernel_only(true);
+    cur_state.set_running_config_only(true);
+    cur_state.retrieve()?;
+
+    for cur_iface in cur_state
+        .interfaces
+        .iter()
+        .filter(|i| i.iface_type() == InterfaceType::Ethernet)
+    {
+        let cur_mac = match cur_iface.base_iface().mac_address.as_ref() {
+            Some(c) => c,
+            None => continue,
+        };
+        if pin_state
+            .interfaces
+            .get_iface(cur_iface.name(), cur_iface.iface_type())
+            .is_none()
+        {
+            for pin_iface in pin_state
+                .interfaces
+                .iter()
+                .filter(|i| i.iface_type() == InterfaceType::Ethernet)
```

Comment:
We need to think about bridge/bonds here right?

Reply:
virtual interface name never changes by kernel or OS upgrade.